episodes/github.md

```diff
@@ -4,80 +4,139 @@ teaching: 5
-or an IDE with support for git like VSCode, 
-you can also use that, but we do not provide instructions for it. 
+![](fig/git-rstudio.png){alt="Initializing version control in RStudio Project"}
 
-::: challenge
-## Exercise: Push your existing code to GitHub using the command line
-On your computer, go to the directory of the project you want to add to GitHub using the terminal (git Bash for Windows). 
-You can use the `cd` command to move into the directory. 
-From here you run the following commands to "connect" your existing project to your repo on GitHub. 
-(This is assuming that you created your repo on GitHub and it is currently empty)
+Then confirm that you want to initialize a new git repository for the project.
+
+'Confirm New Git Repository' and then confirm that you want to restart RStudio.
+
+After restarting RStudio, your project is reopened and a 'Git' Version Control tab is added.
+
+If you click on it, you see the following:
+
+![](fig/git-tab-rstudio.png){alt="The Git tab in an RStudio Project"}
+
+The next step is to add all your files to be “monitored”.
+
+*Note that RStudio automatically adds files you want to ignore in a .gitignore file. You can add certain files (e.g. sensitive data that should not be uploaded) yourself if needed.*
+
+In the Git tab, you can tick the files that you want to upload. Then you click on 'Commit pending changes'.
+
+A new window will open. Here you can add a so-called commit message.
+
+Use something descriptive, in this case for example "Initial commit" and click on 'Commit'.
+
+Now, you want to link to your project on GitHub. The home page of the repository on GitHub includes the URL string we need to identify:
+
+![](fig/github-find-repo-string.png){alt="Where to Find Repository URL on GitHub"}
+
+Make sure to copy the HTTPS link and not the SSH link.
+
+-   **Linux and Mac**
+
+On a linux-based machine (including Mac OS), you should be able to go back to RStudio, and open a new Terminal window:
+
+Go to Tools in the menu bar, select Terminal, and then New Terminal
+
+Then paste in the three commands from the GitHub page, into the terminal window and press enter:
+
+``` bash
+git remote add origin https://github.com/yourusername/my-awesome-project.git
+git branch -M main
+git push -u origin main
+```
+
+Refresh the home page of your repository on GitHub to verify that your code is there.
+
+-   **Windows**
+
+On a windows machine, set RStudio to open Git Bash terminals instead of Command Prompt or Windows Powershell.
+
+Go to Tools in the menu bar and select 'Terminal Options'.
+
+Select Git Bash from the drop down menu, select Apply, then OK.
+
+You can then open a new terminal window by going to 'Tools \> Terminal \> New Terminal'.
+
+In the new terminal window, paste in the three commands from the GitHub page and press enter:
+
+``` bash
+git remote add origin https://github.com/yourusername/my-awesome-project.git
+git branch -M main
+git push -u origin main
+```
```

Comment:
I think this is really terrible that you still have to do this and there is now option to configure this through the RStudio interface. But I think this is the only way to set the origin main branch as upstream branch. If you've done this once you can use the push and pull buttons in the Rstudio git interface